src/index.ts

```diff
@@ -22,20 +24,29 @@ async function run() {
     const startupOptions = core.getInput("startup-options").split(" ");
 
     const helperScript = await matlab.generateScript(workspaceDir, command);
-    const execOptions  = { env: {
-        ...process.env,
-        "MW_MATLAB_BUILDTOOL_DEFAULT_PLUGINS_FCN_OVERRIDE":"ciplugins.github.getDefaultPlugins",
-    }};
+    const execOptions = {
+        env: {
+            ...process.env,
+            "MW_MATLAB_BUILDTOOL_DEFAULT_PLUGINS_FCN_OVERRIDE": "ciplugins.github.getDefaultPlugins",
+        }
+    };
 
     await matlab.runCommand(
         helperScript,
         platform,
         architecture,
-        (cmd,args)=>exec.exec(cmd,args,execOptions),
+        (cmd, args) => exec.exec(cmd, args, execOptions),
         startupOptions
-    );
+    ).finally(() => {
+        try {
+            buildSummary.processAndDisplayBuildSummary();
+        } catch (ex) {
+            console.error('An error occurred while reading the build summary file or adding the build summary table:', ex);
```

Comment:
I still wish that we could catch the error closer to where it's being thrown rather than right here so we could avoid giving this blanket error message.

Reply:
Done